iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -550,7 +550,10 @@ public override async Task StopSasTokenLoopAsync()
 
                 // Await the completion of _refreshLoop.
                 // This will ensure that when StopLoopAsync has been exited then no more token refresh attempts are in-progress.
-                await _refreshLoop.ConfigureAwait(false);
+                if (_refreshLoop != null)
+                {
+                    await _refreshLoop.ConfigureAwait(false);
```

Comment:
Through debugging these tests, I found this sometimes failed with an NRE.

Reply:
Oh, I wonder why.

Reply:
User calls OpenAsync, but it fails, so the call to set the SasTokenRefreshesOn doesn't get called. But the RetryDelegatingHandler still exists, so if the client is closed, we call StopSasTokenLoopAsync on the inner handler.